Nextcloud user guide documentation

marcalcobe opened the pull request:
I created a new section of the user guide documentation related with all the Nextcloud Integration and the features the users might be interested in using.

@psatyal could you please review the documentation from a language perspective?
@wielinde could you please review that the step by step is clear? Do we want to test this docs with someone totally disconnected from the Nextcloud integration design and development to see if it works as intended?

https://community.openproject.org/projects/openproject/work_packages/41752/activity

marcalcobe committed:
Next cloud user guide documentation

I created a new section of the user guide documentation related with all the Nextcloud Integration and the features the users might be interested in using.

marcalcobe commented:
Hi @birthe,

I didn't change the index in the md document for the User Guides as I didn't know which priority I should give to this documentation page and therefore the order of the index.

wielinde commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,87 @@
+
+[Nextcloud](https://nextcloud.com/), the worlds’ most deployed on-premises content collaboration platform, and [OpenProject](https://www.openproject.org/), the leading free and open source project management software, join forces. For both Nextcloud and OpenProject **data sovereignty** and **open source** are an integral part of their business strategy and thus the foundation for their alliance.
+
+OpenProject and Nextcloud together combine project management and content collaboration and complement each other’s services. The collaboration of these two companies offers users a new dimension of data sovereignty and increase in productivity. Users will not risk compliance and data leaks by losing control over their content and can now even easily combine both systems.
```
- I would talk of "communities" instead of "companies"
- I wonder if the focus on compliance and data leaks is the right focus. I always thought the focus was to solve the two questions: "Where are the files I need to for completing the task?" and "Where shall I store the files when I have completed my taks?"

marcalcobe replied:
Hi @wielinde,
This text is mainly extracted from a blog post: https://www.openproject.org/blog/openproject-nextcloud-collaboration/ but yes I agree we can focalise more into this two questions.

psatyal replied:
This part has been changed since :)

wielinde commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,87 @@
+
+OpenProject and Nextcloud together combine project management and content collaboration and complement each other’s services. The collaboration of these two companies offers users a new dimension of data sovereignty and increase in productivity. Users will not risk compliance and data leaks by losing control over their content and can now even easily combine both systems.
+
+The integration is available starting with Nextcloud 20 and OpenProject 12.2. To be able to use this integration the administrator of your instance should have completed the [Nextcloud integration setup](../../system-admin-guide/integrations/nextcloud) configuration.
```
With the release of OP 12.2 the minimal supported version of NC is 22.

marcalcobe replied:
I will modify it

wielinde commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,87 @@
+
+The integration enables users to perform multiple actions in both Nextcloud and OpenProject:
+
+- Keep an eye on ongoing project activities directly in their Nextcloud instance 
```
This is referring to the notifications widget in the dashboard, right? I don't think that this is the most important contribution. I would put the integration between files and work packages first.

marcalcobe replied:
I will modify the order of that.

wielinde commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,87 @@
+- Link their Nextcloud files to OpenProject work packages via the Nextcloud interface.
+- Check which work packages are related to each file in Nextcloud.
+- Have a clear list view of the files linked to a specific work package in OpenProject.
+- Avoid losing control of your content so there is no compliance risks or data leaks.
```
Maybe that needs a bit of explanation? I that because the files are not hosted in other file storages that are not NC?

marcalcobe replied:
Hi @wielinde,
I don't understand what are you referring with this comment? Can you extend what you want to focalise on?

wielinde commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,87 @@
+- Have a clear list view of the files linked to a specific work package in OpenProject.
+- Avoid losing control of your content so there is no compliance risks or data leaks.
+
+## How can I use with this integration?
```
Ok, I see that the text needs still polishing. Maybe I shall wait until @psatyal has gone over the text?

marcalcobe replied:
Yes, @psatyal still needs to review it but anyway your feedback is already valuable

wielinde reviewed (commented):
@marcalcobe I see that the text needs still polishing. Maybe I shall wait until @psatyal has gone over the text?

marcalcobe committed:
Updated the docs with the last feedback

After the feedback provided by @wielinde, I updated and refined the description and  texts.

psatyal committed:
Changed phrasing for style, legibility and grammar

psatyal reviewed (changes requested):
Hi @marcalcobe. Thank you for writing this up. I went through the text and made a few changes (grammar, phrasing). I also removed some sentences I thought were redundant (since the same points were made a bit lower). From a grammar point of view, this looks okay to me, but @wielinde, need your review to make sure that from a content point of view, all important points are covered.

psatyal commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,85 @@
+
+# Nextcloud integration
+
+[Nextcloud](https://nextcloud.com/), the world’s most-deployed on-premises collaborative file storage platform and [OpenProject](https://www.openproject.org/), the leading free and open source project management and collaboration software, join forces. **Data sovereignty** and **open source** are core values that are important to both OpenProject and Nextcloud and which firn the foundational comm ...
```
I reduced this bit because I was afraid it would otherwise sound too press release-y. Also changed how we describe OpenProject following @wielinde's comment.

marcalcobe replied:
Sounds good to me. In any case this will change once we have a common agreement on the description of the Integration between marketing, product team and devs.

psatyal commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,85 @@
+
+This collaboration allows the users to link files and folders in Nextcloud with work packages in OpenProject. Users can now can see all files related to a work package and all work packages related to a file. Project members no longer need to lose time time trying to find the right files to be able to complete their tasks, and the intergatino of OpenProject-specific notifications on Nextcloud das ...
+
+The integration is available starting with Nextcloud 22 and OpenProject 12.2. To be able to use this integration, the administrator of your instance should have completed the [Nextcloud integration setup](../../system-admin-guide/integrations/nextcloud).
```
Are these numbers correct, @marcalcobe and @wielinde ?

marcalcobe replied:
Hi @psatyal,
Yes that are the correct versions.

psatyal commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,85 @@
+
+## Possible errors and troubleshooting
+
+- **OpenProject - No permissions to see the file:** If the user doesn't have permissions to see the details of a file or to open some of the files linked to this work packages (due to permissions, they will be able to see the name, time of last modification and the name of the modifier but will _not_ be able to perform any further actions. To open or access such files, they will need to contact t ...
```
There was a mix of "the user" (third person) and "you" (second person). I've readjusted the text so everything is in third person, but it's a stylistic choice. Can also rewrite for "you" if we prefer the more direct/personal style.

marcalcobe replied:
Hi @psatyal did you review this in all the text? I don't really mind to work with 3rd person but it might got mixed because I used fragments of previous documentation or blogs so makes sense to double check it.

birthe replied:
Hi @marcalcobe and @psatyal, we directly address of the user in 2nd person in our docs. We should try to keep this consistent.

marcalcobe replied:
Hi @birthe,
I will update the documentation to use the 2nd person.

psatyal committed:
Minor changes to align with the set up guide

marcalcobe reviewed (commented):
All the changes has been done on the text. The only pending text would be the unified description of the integration in case is needed on the top of the documentation page.
Also a review from any @opf/doc-writers would be appreciated as double check.

HDinger commented:
I think the base branch should probably be `release/12.2`, right? If you want to, we can shortly chat about that.

marcalcobe committed:
Review language (2nd person) and styles

Double check that everything is in 2nd person and double check some styles and paragraphs that where not 100% clear.

birthe commented:
> I think the base branch should probably be `release/12.2`, right? If you want to, we can shortly chat about that.

Hi @HDinger, we currently merge changes of the docs which are not yet released to the Dev branch - to have the possiblity to preview the changes on our Edge environment. @as-op said he will take care to cherry pick all changes over to release :-)

psatyal commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -0,0 +1,87 @@
+
+# Nextcloud integration
+
+[Nextcloud](https://nextcloud.com/), the world’s most-deployed on-premises collaborative file storage platform and [OpenProject](https://www.openproject.org/), the leading free and open source project management and collaboration software, join forces. **Data sovereignty** and **open source** are important core values to both OpenProject and Nextcloud and they form the foundational common ground  ...
```
I think perhaps we should change "join forces'; this could imply some form of merger.

psatyal committed:
Rereviewed the text a second time and made changes

Made changes to language, grammar and corrected errors that I hadn't noticed the first time around

psatyal reviewed (approved):
Hi @birthe, @as-op and @marcalcobe. I went through the text one more time and made a few edits (the usual: phrasing, grammar, types...). I also corrected a few things I hadn't caught the first time around. It looks good to me now.

marcalcobe commented:
Hi @as-op and @birthe,
I think this PR is also ready to move to 12.2 and merge.

as-op commented:
Recreated & merged to release/12.2 at https://github.com/opf/openproject/pull/11063
